realm/realm-library/src/main/cpp/io_realm_internal_Collection.cpp

```diff
@@ -433,3 +433,19 @@ Java_io_realm_internal_Collection_nativeGetMode(JNIEnv *env, jclass, jlong nativ
     return -1; // Invalid mode value
 }
 
+JNIEXPORT jlong JNICALL
+Java_io_realm_internal_Collection_nativeCreateResultsFromBacklinks(JNIEnv *env, jclass, jlong shared_realm_ptr, jlong row_ptr, jlong src_table_ptr, jlong src_col_index) {
+    TR_ENTER_PTR(row_ptr)
+    if (!ROW_VALID(env, ROW(row_ptr)))
+        return -1;
+    try {
+        Row* row = ROW(row_ptr);
+        Table* src_table = TBL(src_table_ptr);
```

Comment:
I will recommend to use `auto` and `reinterpret_cast` instead of these old C macros.